packages/renderer/src/stores/informerWritable.ts

```diff
@@ -0,0 +1,39 @@
+import type { Invalidator, StartStopNotifier, Subscriber, Unsubscriber, Writable } from 'svelte/store';
+import { writable } from 'svelte/store';
+
+/** Writable interface for both updating and subscribing. */
+export interface InformerWritable<T> extends Writable<T> {
+  /**
+   * Get the active informer id
+   */
+  getInformerId(): number | undefined;
+}
+
+export function customWritable<T>(
+  value: T,
+  startInformer?: () => Promise<number>,
+  start?: StartStopNotifier<T>,
+): InformerWritable<T> {
+  let informer: number | undefined;
+  const origWritable = writable(value, start);
+
+  function subscribe(this: void, run: Subscriber<T>, invalidate?: Invalidator<T>): Unsubscriber {
+    startInformer?.()
+      .then(id => (informer = id))
+      .catch((e: unknown) => console.error(e));
```

Comment:
might be good to say in the error message some context to better understand the error (let say we only see in the console `invalid name` or `parsing error` it's hard to know what happened